tests/monotouch-test/CoreFoundation/OSLogTest.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.CoreFoundation {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class OSLogTest {
+
+		[Test]
+		public void Default ()
+		{
+			OSLog.Default.Log (OSLogLevel.Default, "monotouch-test / Default / Default");
+			// this will show in the application output (e.g. inside VSfM)
+		}
+
+		[Test]
+		public void Custom ()
+		{
+			using (var log = new OSLog ("subsystem", "category")) {
+				log.Log (OSLogLevel.Error, "monotouch-test /  custom / Debug");
+				// this will show in the application output (e.g. inside VSfM)
+				// and also inside Console.app under the simulator/device
```

Comment:
No Asserts?

Reply:
As long as it does to crash or throw here :P I think this is fine

Reply:
@mandel-macaque not quite sure on what to assert (if it does not crash, like it would without the fix). Reading console.app is a quite troublesome from simulator and would not work on devices. Got any ideas ?

Reply:
I think there is a way with http://www.cplusplus.com/reference/cstdio/freopen/ freopen stderr to a file, read wha was logged in the file. Maybe too much work :/

Reply:
I'm fine with "not crashing is good enough". I could see redirects being brittle maybe.

Reply:
I won't stop you PR'ing that... but, for myself, messing with stderr, which is already used used by the test harness, seems more a problem than a solution :)

Reply:
There's a precedent for redirecting stderr... https://github.com/xamarin/maccore/blob/4c92d05de776e6166648f57ff26a5efab8c06363/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/InstallationProxyService.cs#L95-L109